components/camel-mongodb3/src/main/java/org/apache/camel/component/mongodb3/MongoDbEndpoint.java

```diff
@@ -302,8 +303,12 @@ public Exchange createMongoDbExchange(Document dbObj) {
     
     @Override
     protected void doStart() throws Exception {
-        mongoConnection = CamelContextHelper.mandatoryLookup(getCamelContext(), connectionBean, MongoClient.class);
-        log.debug("Resolved the connection with the name {} as {}", connectionBean, mongoConnection);
+        if( mongoConnection == null ) {
+            mongoConnection = CamelContextHelper.mandatoryLookup(getCamelContext(), connectionBean, MongoClient.class);
+            log.debug("Resolved the connection provided by {} context reference as {}", connectionBean, mongoConnection);
+        }else {
```

Comment:
I think this will give an error on the codestyle. Can you run the sourcecheck profile and fix the errors? Thanks